Main calendar functionality

rindo789 opened the pull request:
Created functionality for the Main Calendar to be able to open all existing activities from different apps with a registered calendar and to be able to create activities for apps where a calendar is registered. Also added a new Activity model where a user will be able to create a regular activity. #68

rindo789 committed:
main calendar functionality

wai-blue commented on src/core/Calendar.php:
```diff
@@ -5,6 +5,7 @@
 class Calendar implements \ADIOS\Core\Testable {
 
   public \HubletoMain $main;
+  public array $formComponent = [];
```
- Rename to activitySelectorSomething
- Add PHP stan array definition
- Add a comment explaining how it should be used and where it is used

rindo789 committed:
class attribute name change and fixes

rindo789 commented:
- changed the `formComponent` class attribute name to `activitySelectorConfig`
  - changed the `form` key-pair to `formComponent`
- added a description to the `activitySelectorConfig` class attribute and a PHPStan specification
- added a onSaveCallback function when opening an already existing activity
